Sources/Commands/SwiftPackageTool.swift

```diff
@@ -1036,6 +1036,9 @@ extension SwiftPackageTool {
                     }
                 }
             }
+            
+            // Pass any targets that were specified, followed by any plugin arguments that were given on the command line.
+            let pluginArguments = targetNames.flatMap{ ["--target", $0] } + pluginCommand.dropFirst()
```

Comment:
do products need special treatment also?

Reply:
Yes, I'll be adding that, and also removing the fallback of passing all targets.  I don't think there are other flags that it makes sense to propagate — and that is something we can add to over time based on toolsversion without breaking anyone.  I think that's the last thing to fix before putting this out of draft.